indexing-service/src/main/java/org/apache/druid/indexing/common/TaskLock.java

```diff
@@ -62,4 +63,17 @@ public interface TaskLock
   boolean isRevoked();
 
   boolean conflict(LockRequest request);
+
+  /**
+   * Checks if the lock is revoked and throws a {@link LockAcquisitionFailedException} if so.
+   *
+   * @param interval The time interval for which the lock is acquired.
+   * @throws LockAcquisitionFailedException if the lock is revoked.
+   */
+  default void assertNotRevoked(Interval interval)
+  {
+    if (isRevoked()) {
+      throw new LockAcquisitionFailedException(interval);
```

Comment:
I think it would be better to throw a `DruidException` instead.

Reply:
Done!

Reply:
One of the reason a new exception type was introduced is to distinguish terminal and intermittent task failures.
In this case a lock acquisition failed exception is a intermittent failure means the operator not require to take action to fix this.
If use the more general `DruidException`, though can look at the exception message to know the cause but that's easy to break as a minor _improvement_ on the message can break the checking logic external of Druid code.

Reply:
`DruidException` is actually fine for now. The only real external consumer is us and I anticipate a lot of this code to be reworked in the coming future.